### PR TITLE
Add remote-script template

### PR DESCRIPTION
**Pull request by drasmuss**
**Motivation and context:**

Something that has come up a lot is that we want to run one of the CI scripts on some remote machine. This script builds on top of the more general `remote.sh` template to make that specific use case easier.

**Interactions with other PRs:**

NengoDL, KerasSpiking, and NengoFPGA all currently have their own versions of this script; we should switch them over to using this template (and make sure that this script covers all those use cases).

**How has this been tested?**

Added some tests on the content of the rendered script, as well as a new functional job which runs the test script on a remote.

**How long should this take to review?**

- Average (neither quick nor lengthy)

**Types of changes:**

- New feature (non-breaking change which adds functionality)

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [x] I have included a changelog entry.
- [x] I have added tests to cover my changes.
- [x] I have run the test suite locally and all tests passed.

### CODE REVIEW

File: nengo_bones/templates/remote-script.sh.template [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        echo "Waiting for lock on device $DEVICE_ID"
+        (
+            flock -x -w 540 200 || exit 1
+            bash .ci/{{ remote_script }}.sh script || exit 1
+        ) 200>/var/lock/.travis-ci.exclusivelock."$DEVICE_ID" || REMOTE_STATUS=1
+{% endblock %}
+
+{% block after_script %}
+    {% if remote_script == "docs" %}
```
Comment (hunse) [inline nengo_bones/templates/remote-script.sh.template:64]: Just a personal preference thing, but I would un-indent these by two spaces just to make it easier to distinguish the templating from the actual code in this block.

Comment (tbekolay) [reply nengo_bones/templates/remote-script.sh.template:48]: I'm pretty indifferent (the Jinja templates don't have a well-specified style) but I'm going to leave it as is to be consistent with the other templates in this repo.

**Review by tbekolay — APPROVED** [review]
I made a small fixup to replace the `awk` command with `cut`. The only other thought that I had was that a bunch of things will break if a CI script has a space in it, which we could add a check for somewhere if we're worried about it. I'm not worried for the repos we control now, but new repos/teams/external users could get tripped up with it :shrug:

Anyway if the fixup looks good to you @drasmuss I'll squash and merge.

Update: to speed things up (travis-ci.org is super slow) I did the squash already; the line I changed was [this one](https://github.com/nengo/nengo-bones/pull/124/commits/e7c63df022baa9ea5ff5453c6f8bdd75489c3c45#diff-aeeba8e1bc1b36bc682c34f7e0d550c433759668a66dfcd9d290177812600a8fR19)

**Comment by drasmuss** [comment]
I used awk because it allows us to get the last element in a series of unknown length, because I'm not sure that it'll always be length 2. E.g., when doing pip freeze I've seen nengo-bones @ git+https:..., but also just -e git+https://. It's probably relatively reliable here since we're always using the same environment and installing in the same way, but it doesn't hurt to be careful I think unless you have a strong reason for avoiding awk.

**Comment by tbekolay** [comment]
> I'm not sure that it'll always be length 2

Ah, didn't realize that the `pip freeze` lines could be different. Generally I find `awk` commands inscrutable but `cut` can't get the last field reliably so I'll revert :+1:

**Pull request merged**